open-api/rest-catalog-open-api.yaml

```diff
@@ -1444,7 +1444,7 @@ components:
       schema:
         type: string
       required: true
-      description: An optional prefix in the path
+      description: Prefix in the path
```

Comment:
It is actually a required field. Cannot be optional. 

I tried changing the field to optional. I get validation error as  "in: path" params must be required. 
So updated the description.

Reply:
so should required be set to false then?

Reply:
What's the error you're seeing?

The prefix is optional, but it depends on the definition. We can also update the default to be an empty string.

Reply:
If I change to `required: false` for `prefix`

```
Spec is invalid.
Issues:

	components.parameters.For path parameter prefix the required value should be true


Execution failed for task ':iceberg-open-api:validateRESTCatalogSpec'.
> Validation failed.
```

Reply:
I think if we need optional params in path, we need to define a separate path without prefix.

Reply:
I think it makes sense to have `prefix` optional.  Specifically for the `prefix` I would set `required: false` and check in the validator.

Reply:
validator doesn't allow setting `required` as `false` for the Path parameter.

What we need maybe have two paths defined, one with prefix and one without prefix?

Reply:
Wouldn't it be confusing for users ? Which path should I use ?

Reply:
Optional path parameters are not allowed in the V3 spec: https://swagger.io/docs/specification/describing-parameters/ 
Let's not duplicate the paths since that's going to be confusing as JB pointed out. There is a long discussion going on here: https://github.com/OAI/OpenAPI-Specification/issues/93#issuecomment-1908673473 And it looks like it gets support in V4. I would say that we wait until that's being released and solve it then.

Reply:
Thanks @Fokko digging more on this. Their goal for publishing V4 is end of 2024. 

So, As of now (since spec is based on v3), it is a required field. So, lets update the description and merge this as it is confusing.

Reply:
I agree, let's keep as it is for now. As we have the new REST spec proposal, we can add a topic related to that.